.github/scripts/create-lambda-function.sh

```diff
@@ -0,0 +1,169 @@
+CANADA_CA_SEARCH_API_KEY=$(echo "$PARAMETERS_JSON" | jq -r '.[] | select(.Name=="canada_ca_search_api_key") | .Value')
+JWT_SECRET_KEY=$(echo "$PARAMETERS_JSON" | jq -r '.[] | select(.Name=="jwt_secret_key") | .Value')
+USER_AGENT=$(echo "$PARAMETERS_JSON" | jq -r '.[] | select(.Name=="user_agent") | .Value')
+GOOGLE_API_KEY=$(echo "$PARAMETERS_JSON" | jq -r '.[] | select(.Name=="google_api_key") | .Value')
+GOOGLE_SEARCH_ENGINE_ID=$(echo "$PARAMETERS_JSON" | jq -r '.[] | select(.Name=="google_search_engine_id") | .Value')
+
+# Validate that all required parameters were extracted
+if [ -z "$DOCDB_URI" ] || [ -z "$AZURE_OPENAI_API_KEY" ] || [ -z "$AZURE_OPENAI_ENDPOINT" ] || [ -z "$JWT_SECRET_KEY" ]; then
+  echo "Error: One or more required parameters are empty after extraction"
+  echo "DOCDB_URI: ${DOCDB_URI:+SET}"
+  echo "AZURE_OPENAI_API_KEY: ${AZURE_OPENAI_API_KEY:+SET}"
+  echo "AZURE_OPENAI_ENDPOINT: ${AZURE_OPENAI_ENDPOINT:+SET}"
+  echo "JWT_SECRET_KEY: ${JWT_SECRET_KEY:+SET}"
+  exit 1
+fi
+
+# Get VPC configuration for Lambda
+echo "Fetching VPC configuration for Lambda..."
+VPC_ID="$(aws ec2 describe-vpcs --filters "Name=tag:Name,Values=ai-answers_vpc" --query 'Vpcs[0].VpcId')"
+if [ "$VPC_ID" = "None" ] || [ -z "$VPC_ID" ]; then
+  echo "Error: Could not find ai-answers VPC"
+  exit 1
+fi
+
+# Get private subnet IDs
+SUBNET_IDS_RAW="$(aws ec2 describe-subnets --filters "Name=vpc-id,Values=$VPC_ID" "Name=tag:Name,Values=*private*" --query 'Subnets[].SubnetId' --output text)"
+# Convert tab-separated list to comma-separated list for Lambda
+SUBNET_IDS=$(echo "$SUBNET_IDS_RAW" | tr '\t' ',')
+if [ -z "$SUBNET_IDS" ]; then
+  echo "Error: Could not find private subnets in VPC $VPC_ID"
+  exit 1
+fi
+
+# Get the Lambda security group ID
+LAMBDA_SG_ID=$(aws ec2 describe-security-groups --filters "Name=vpc-id,Values=$VPC_ID" "Name=group-name,Values=ai-answers-lambda-pr-review" --query 'SecurityGroups[0].GroupId' --output text)
+if [ "$LAMBDA_SG_ID" = "None" ] || [ -z "$LAMBDA_SG_ID" ]; then
+  echo "Warning: Lambda security group not found, Lambda will run without VPC configuration"
+  VPC_CONFIG=""
+else
+  echo "Found VPC configuration (VPC and security group located)"
+  VPC_CONFIG="--vpc-config SubnetIds=$SUBNET_IDS,SecurityGroupIds=$LAMBDA_SG_ID"
+fi
+
+# Check if the function already exists
+if aws lambda get-function --function-name "$FULL_FUNCTION_NAME" > /dev/null 2>&1; then
+  echo "Function exists. Updating code and environment variables..."
+  
+  echo "Updating function code..."
+  if ! aws lambda update-function-code \
+    --function-name "$FULL_FUNCTION_NAME" \
+    --image-uri "${REGISTRY}/${IMAGE}:${IMAGE_TAG}" > /dev/null 2>&1; then
+    echo "Error: Failed to update function code"
+    exit 1
+  fi
+  
+  echo "Waiting for function to be ready for configuration update..."
+  aws lambda wait function-updated --function-name "$FULL_FUNCTION_NAME" 2>/dev/null || true
```

Comment:
I think this should wait (or fail):
```suggestion
  aws lambda wait function-updated --function-name "$FULL_FUNCTION_NAME" 2>/dev/null
```